x-pack/platform/plugins/shared/observability_ai_assistant/common/types.ts

```diff
@@ -29,38 +29,65 @@ export interface PendingMessage {
   aborted?: boolean;
   error?: any;
 }
-export interface DetectedEntity {
-  entity: string;
-  class_name: string;
-  start_pos: number;
-  end_pos: number;
-  hash: string;
-  type: 'ner' | 'regex';
+
+export interface Deanonymization {
+  start: number;
+  end: number;
+  entity: {
+    class_name: string;
+    value: string;
+    mask: string;
+  };
 }
 
-export type DetectedEntityType = DetectedEntity['type'];
-export interface Unredaction {
-  entity: string;
-  class_name: string;
-  start_pos: number;
-  end_pos: number;
-  type: 'ner' | 'regex';
+export interface DeanonymizationItem {
+  message: {
+    role: MessageRole;
+    content?: string;
+    toolCalls?: Array<{
+      function: {
+        name: string;
+        arguments: Record<string, any> | {};
+      };
+    }>;
+    name?: string;
+    response?: Record<string, any>;
+    toolCallId?: string;
+  };
+  deanonymizations: Deanonymization[];
 }
 
-export type UnredactionType = Unredaction['type'];
+export type DeanonymizationInput = DeanonymizationItem[];
+
+export interface DeanonymizationOutput {
+  message: {
+    content?: string;
+    toolCalls?: Array<{
+      toolCallId: string;
+      function: {
+        name: string;
+        arguments: Record<string, any>;
+      };
+    }>;
```

Comment:
I get confused about the data modeling here. I think this should either just be the output from the inference client, or it should be our own model (which means single function_call).

Reply:
Isn't that the output from the inference client?  array of `toolCalls`.